src/build_and_deploy.sh

```diff
@@ -0,0 +1,106 @@
+#
+# The `build_and_deploy.sh` file centralizes functions related to kw bd.
+#
+
+include "${KW_LIB_DIR}/deploy.sh"
+include "${KW_LIB_DIR}/build.sh"
+
+# kw bd is a shortcut to build and deploy a new kernel.
+# It also supports arguments related to git bisect usage.
+#
+# The idea of using git bisect along with kw bd is to find the
+# first commit where building and/or deploying the kernel failed.
+#
+# @raw_options String with all user options
+#
+function build_and_deploy_main()
+{
+
+  local raw_options="$*"
+
+  if [[ "$1" =~ -h|--help ]]; then
+    build_and_deploy_help "$1"
+    exit 0
+  fi
+
+  last_commit=$(eval git rev-parse HEAD)
+  parse_build_and_deploy_options "$last_commit" "$raw_options"
+
+  if [[ "$?" -gt 0 ]]; then
+    complain "${options_values['ERROR']}"
+    exit 22 # EINVAL
+  fi
+
+  run_bisect="${options_values['GOOD_OR_BAD']}"
+  good="${options_values['GOOD']}"
+  bad="${options_values['BAD']}"
+  deploy_main_args="${options_values['DEPLOY_MAIN_ARGS']}"
+
+  if [[ $run_bisect == 0 ]]; then
+    eval "build_kernel_main && deploy_main 1 ""$deploy_main_args"""
```

Comment:
Why do you need to eval here? Since you created a new file, it is better if you handle each function and check the output of each command.